### PR TITLE
Move and improve async snippets

### LINKED ISSUE #16987: Missing function definition
Could you please provide a definition of ToastBreadAsync()? Thank you.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 17d27a0d-c072-8c9f-7e31-3de60bee9ae8
* Version Independent ID: e0c09794-e712-19ff-cb9f-12648d60d4f6
* Content: [Asynchronous programming in C#](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/#feedback)
* Content Source: [docs/csharp/programming-guide/concepts/async/index.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/concepts/async/index.md)
* Product: **dotnet-csharp**
* Technology: **csharp-async**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #18477: Using List.Length instead of List.Any()
Do you recommend using allTasks.Any() instead of allTasks.Length &gt; 0? I think that documentation must provide the right path to write code in programming guide. Why not to use Length where it is possible? People learning this languages then face the job interviews and failed at this point.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 17d27a0d-c072-8c9f-7e31-3de60bee9ae8
* Version Independent ID: e0c09794-e712-19ff-cb9f-12648d60d4f6
* Content: [Asynchronous programming in C#](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/#feedback)
* Content Source: [docs/csharp/programming-guide/concepts/async/index.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/concepts/async/index.md)
* Product: **dotnet-csharp**
* Technology: **csharp-async**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #18477
Fixes #16987

### CODE REVIEW

File: docs/csharp/programming-guide/concepts/async/snippets/index/AsyncBreakfast-final/Program.cs [hunk 1]
```diff
@@ -18,7 +18,7 @@ static async Task Main(string[] args)
 
             // <SnippetAwaitAnyTask>
             var allTasks = new List<Task>{eggsTask, baconTask, toastTask};
-            while (allTasks.Any())
+            while (allTasks.Count > 1)
```
Comment (pkulikov) [inline docs/csharp/programming-guide/concepts/async/snippets/index/AsyncBreakfast-final/Program.cs:21]: Looks like it doesn't wait for the last task. Should it compare with zero?

Comment (Youssef1313) [reply docs/csharp/programming-guide/concepts/async/snippets/index/AsyncBreakfast-final/Program.cs:21]:
```suggestion
            while (allTasks.Count > 0)
```

**Review by IEvangelist — CHANGES_REQUESTED** [review]
In the https://github.com/Youssef1313/docs/blob/fix-18477/docs/csharp/programming-guide/concepts/async/snippets/async-return-types/async-valuetask.cs file, could we make the main method `Task` based - and make the `Roll` and `GetDiceRoll` methods adhere to the `Async` suffix?

```cs
static async Task Main()
{
    Console.WriteLine($"You rolled {await GetDiceRollAsync()}");
}
```

**Comment by BillWagner** [comment]
I'm ready to :shipit: once the changes requested by @IEvangelist are addressed.

Thanks for making these updates @Youssef1313

**Comment by BillWagner** [comment]
Ping @IEvangelist The requested changes have been made.

**Review by IEvangelist — APPROVED** [review]
Looks good to me :shipit:

**Pull request merged**